### PR TITLE
updated to pfunit-4.0

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
changes for pFunit 4.0

### CODE REVIEW

**Comment by mathomp4** [comment]
@weiyuan-jiang I think we might want to do this differently.

First, the bits in `FindBaselibs.cmake` should probably just become:
```cmake
find_package(PFUNIT QUIET)
```
Then I think in `esma.cmake` we might not even need the `OPTION` but instead could do:
```cmake
if (PFUNIT_FOUND)
     add_custom_target(tests COMMAND ${CMAKE_CTEST_COMMAND})
￼endif ()
```
I'll let @tclune weigh in more. Maybe not QUIET? Not sure...

File: esma.cmake [hunk 1]
```diff
@@ -83,6 +83,7 @@ option (ESMA_ALLOW_DEPRECATED "suppress warnings about deprecated features" ON)
 # Unit testing
 option (PFUNIT "Activate pfunit based tests" OFF)
 if (PFUNIT)
+   find_package(PFUNIT REQUIRED)
```
Comment (tclune) [inline esma.cmake]: I agree with Matt's suggestion.     That's the way I do it in my other projects.   

We do want to make certain that none of the tests are added to the "all" target though, as everyone will have PFUNIT found via baselibs going forward and not everyone wants to build the tests, and not everyone would even understand what they are for.

**Comment by weiyuan-jiang** [comment]
Now the FindBaselibs will check if PFUNIT is on and if it is found

**Comment by mathomp4** [comment]
Hmm. It still might be too much there. This:
```cmake
find_package(PFUNIT QUIET)
```
does this:
```cmake
set (PFUNIT_PATH ${BASEDIR}/PFUNIT-4.0)
set (PFUNIT_INCLUDE_DIRS ${PFUNIT_PATH}/mod ${PFUNIT_PATH}/include)		￼      
set (PFUNIT_LIBRARY_DIRS ${PFUNIT_PATH}/lib)
set (PFUNIT_LIBRARIES ${PFUNIT_PATH}/lib/libpfunit.a)
set (PFUNIT_INCLUDE_DIRS ${PFUNIT_PATH}/include)
```
With `find_package()` you should no longer need any of that.

The next main question is if we need these bits:
```cmake
option (PFUNIT "Activate pfunit based tests" OFF)
if (PFUNIT)
endif (PFUNIT)
```
at all either. If PFUNIT_FOUND is true, then you can build the tests if `make tests` is called. But, as @tclune says, probably need figure out how to apply an `EXCLUDE_FROM_ALL` so we don't build the tests at all times.

**Comment by weiyuan-jiang** [comment]
If PFUNIT_FOUND is true, then tests are build only if make tests is called after applying EXCLUDE_FROM_ALL to add_subdirectory

**Pull request merged**